lib/api/contacts.rb

```diff
@@ -12,14 +12,10 @@ def hash_access
     end
 
     def params
-      super
-    end
-
-    def opts
-      '&property=vid&property=firstname&property=lastname&'\
+      super({property: '&property=vid&property=firstname&property=lastname&'\
```

Comment:
Looks like I missed a self.opts there.  I think it should be this on `Api::Base` :
```ruby
def retreive
  options = self.respond_to?(:opts) ? self.opts : nil
```

and then change line 15 of `Rest::Api` to:
```ruby
uri.query += opts if !opts.nil?
```